src/threadpool_scheduler.cpp

```diff
@@ -64,15 +69,77 @@ struct threadpool_data {
 	std::condition_variable shutdown_complete_event;
 #endif
 };
-
+    
+#if defined(EMULATE_PTHREAD_THREAD_LOCAL)
+static pthread_key_t owning_threadpool_key;
+static pthread_key_t thread_id_key;
```

Comment:
Use a single `pthread_key_t` which points to an allocated piece of memory containing both values. Calls to `pthread_getspecific` are rather expensive.